### PR TITLE
[BugFix] fix qwen3.5+pcp+mtp error

### PR DESCRIPTION
**Pull request by weiguihua2**
### What this PR does / why we need it?
The calculation logic for qwen3.5 with pcp and mpt logits superimposed has been modified.

After the repair, 50 GPQA runs were performed, with an accuracy of 90%. The per-position acceptance rates were: 0.886, 0.739, and 0.616, and the average draft acceptance rate was 74.7%.

[2026-06-26 15:29:38,507] [ais_bench] [INFO] Task vllm-api-general-chat/GPQA_diamond: {'accuracy': np.float64(90.0), 'type': 'GEN'}
[2026-06-26 15:29:38,523] [ais_bench] [INFO] Evaluation task time elapsed: 0.09s
[2026-06-26 15:29:39,781] [ais_bench] [INFO] Evaluation tasks completed.
[2026-06-26 15:29:39,784] [ais_bench] [INFO] Summarizing evaluation results...
dataset       version    metric    mode      vllm-api-general-chat

GPQA_diamond  b1ed2c     accuracy  gen                       90.00

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/967c5c3bc38891f4465d3f4e99917ed837bb3833

### CODE REVIEW

**Review by pisceskkk — APPROVED** [review]
LGTM

**Review by zhangxinyuehfad — APPROVED** [review]

**Review by MengqingCao — APPROVED** [review]
